bin/pre-commit-hook.js

```diff
@@ -40,46 +27,24 @@ require( '../server/config/validate-config-keys' );
  * @returns {Array}          Paths output from git command
  */
 function parseGitDiffToPathArray( command ) {
-	return execSync( command )
-		.toString()
+	return execSync( command, { encoding: 'utf8' } )
 		.split( '\n' )
 		.map( name => name.trim() )
-		.filter(
-			name => name.endsWith( '.js' ) || name.endsWith( '.jsx' ) || name.endsWith( '.scss' )
-		);
+		.filter( name => /\.(jsx?|scss)$/.test( name ) )
 }
 
 const dirtyFiles = new Set( parseGitDiffToPathArray( 'git diff --name-only --diff-filter=ACM' ) );
-
 const files = parseGitDiffToPathArray( 'git diff --cached --name-only --diff-filter=ACM' );
 
-// run prettier for any files in the commit that have @format within their first docblock
-files.forEach( file => {
-	const text = fs.readFileSync( file, 'utf8' );
-	if ( shouldFormat( text ) ) {
-		// File has unstaged changes. It's a bad idea to modify and add it before commit.
-		if ( dirtyFiles.has( file ) ) {
-			console.log(
-				chalk.red( `${ file } will not be auto-formatted because it has unstaged changes.` )
-			);
-			return;
-		}
+dirtyFiles.forEach( file => console.log(
+	chalk.red( `${ file } will not be auto-formatted because it has unstaged changes.` )
+) );
 
-		const formattedText = prettier.format(
-			text,
-			file.endsWith( '.scss' ) ? sassPrettierConfig : defaultPrettierConfig
-		);
+const toPrettify = files.filter( file => ! dirtyFiles.has( file ) );
+toPrettify.forEach( file => console.log( `Prettier formatting file: ${ file } because it contains the @format flag` ) );
 
-		// No change required.
-		if ( text === formattedText ) {
-			return;
-		}
-
-		fs.writeFileSync( file, formattedText );
-		console.log( `Prettier formatting file: ${ file } because it contains the @format flag` );
-		execSync( `git add ${ file }` );
-	}
-} );
+execSync( `npx prettier --write --requirePragma ${ toPrettify.join( ' ' ) }` );
```

Comment:
We probably want to make this the following:

```bash
npx --no-install prettier --write --require-pragma \"**/*.{js,jsx,scss}\"
```

`--no-install` will ensure that `npx` doesn't run off and install the npm published `prettier` package instead of our fork.

Reply:
Same here, if `prettier` is installed, then `execSync( './node_modules/.bin/prettier --write ...' )` is more straight-forward.